### PR TITLE
PreflightGuide: new section on the leader schedule

### PR DESCRIPTION
**Pull request by amesgen**
Partially addresses #887

### CODE REVIEW

**Comment by amesgen** [comment]
[Rendered](https://github.com/IntersectMBO/ouroboros-consensus/blob/amesgen/leader-schedule-stats-docs/docs/website/contents/for-developers/PreflightGuide.md#dynamics-of-the-leader-schedule)

File: docs/website/contents/for-developers/PreflightGuide.md [hunk 1]
```diff
@@ -128,6 +128,124 @@ Use a ledger snapshot of your local node (make sure to add the `_db-analyser` su
...
+> which yields 0.0185%, or the probability that it takes more than one minute until a slot is active
+> ```python
+> stats.geom(1/20).sf(60)
+> ```
+> which yields 4.6%.
+> Calculate some other probabilities that seem interesting to you, maybe inspired by some of the empirical data gathered via db-analyser.
+
+### Multi leader slots
+
+The probability that a pool with relative stake $\sigma$ is elected in any specific slot is given by $ \phi_f(\sigma) = \phi(\sigma) = 1-{(1-f)}^\sigma $.
```
Comment (jasagredo) [inline docs/website/contents/for-developers/PreflightGuide.md:188]: This formula is not rendered
```suggestion
The probability that a pool with relative stake $\sigma$ is elected in any specific slot is given by $\phi_f(\sigma) = \phi(\sigma) = 1-{(1-f)}^\sigma$.
```

File: docs/website/contents/for-developers/PreflightGuide.md [hunk 2]
```diff
@@ -128,6 +128,124 @@ Use a ledger snapshot of your local node (make sure to add the `_db-analyser` su
...
+The discussion above completely ignored *grinding*.
+In the context of the leader schedule, this refers to an adversarial influence on the nonce that is used as a seed for all of the VRF computations for slot election within a specific epoch.
+The exact mechanism is out-of-scope for this section, but it should be said that such an attack requires both a considerable amount of stake and computational resources, and it allows an adversary to choose the best out of multiple epoch nonces.
+An extremely conservative upper bound for how many nonces an attacker can choose from is $10^{20}$ (based on the computational resources required to even compute all of these nonces).
+The mainnet parameters of Cardano (in particular the security parameter) were chosen with resistance against grinding in mind.
+
+Concretely, an attacker with $\alpha$ stake that can choose out of $N$ epoch nonces can ensure that it leads any specific slot with probability $1-{(1-\phi(\alpha))}^N$, which is a huge advantage compared to $\phi(\alpha)$.
+For example, an attacker with $N=10^{20}$ with even a very small amount of stake has a chance of close to 100% to cause themselves to be elected in any specific slot.
```
Comment (jasagredo) [inline docs/website/contents/for-developers/PreflightGuide.md:236]: I understand this as "An attacker can choose (with close to 100% probability) a slot in which they want to be elected" but this doesn't mean that they can choose with 100% probability to be elected in say 5 specific slots in the epoch right? The 100% applies if a single slot is targeted no?

**Review by jasagredo — COMMENTED** [review]
Awesome write-up, 10/10 :ok_hand:

File: docs/website/contents/for-developers/PreflightGuide.md (hunk 2 above)
Comment (amesgen) [reply docs/website/contents/for-developers/PreflightGuide.md:236]: It probably also still applies to 5 slots, but not to large numbers of slots, the next paragraph is also supposed to point this out (slightly modified in the latest commit)

> Usually, an attacker wants to maximize their total number of active slots within a larger number of slots, like in an entire epoch. For this, the impact of grinding is smaller (but still large), as every epoch nonce affects the entire leader schedule of the epoch, and the attacker can not choose different epoch nonces for different slots.

Does that make sense, or do we want to make this more explicit?

Comment (jasagredo) [reply docs/website/contents/for-developers/PreflightGuide.md:236]: I think just modifying this word would be enough:
```suggestion
For example, an attacker with $N=10^{20}$ with even a very small amount of stake has a chance of close to 100% to cause themselves to be elected in a specific slot.
```

```suggestion
For example, an attacker with $N=10^{20}$ with even a very small amount of stake has a chance of close to 100% to cause themselves to be elected in any single specific slot.
```

Comment (jasagredo) [reply docs/website/contents/for-developers/PreflightGuide.md:236]: But also feel free to ignore this comment :smile:

**Review by jasagredo — APPROVED** [review]

**Pull request merged**